src/_P088_HeatpumpIR.ino

```diff
@@ -43,47 +43,9 @@
  *
  */
 
-#include <AUXHeatpumpIR.h>
-#include <BalluHeatpumpIR.h>
-#include <CarrierHeatpumpIR.h>
-#include <DaikinHeatpumpIR.h>
-#include <FuegoHeatpumpIR.h>
-#include <FujitsuHeatpumpIR.h>
-#include <GreeHeatpumpIR.h>
-#include <HisenseHeatpumpIR.h>
-#include <HitachiHeatpumpIR.h>
-#include <HyundaiHeatpumpIR.h>
-#include <IVTHeatpumpIR.h>
-#include <MideaHeatpumpIR.h>
-#include <MitsubishiHeatpumpIR.h>
-#include <MitsubishiHeavyFDTCHeatpumpIR.h>
-#include <MitsubishiHeavyHeatpumpIR.h>
-#include <MitsubishiMSCHeatpumpIR.h>
-#include <MitsubishiSEZKDXXHeatpumpIR.h>
-#include <PanasonicCKPHeatpumpIR.h>
-#include <PanasonicHeatpumpIR.h>
-#include <SamsungHeatpumpIR.h>
-#include <SharpHeatpumpIR.h>
-#include <ToshibaDaiseikaiHeatpumpIR.h>
-#include <ToshibaHeatpumpIR.h>
-
-// Array with all supported heatpumps
-HeatpumpIR *heatpumpIR[] = {new PanasonicCKPHeatpumpIR(), new PanasonicDKEHeatpumpIR(), new PanasonicJKEHeatpumpIR(),
-                            new PanasonicNKEHeatpumpIR(), new PanasonicLKEHeatpumpIR(),
-                            new CarrierNQVHeatpumpIR(), new CarrierMCAHeatpumpIR(),
-                            new MideaHeatpumpIR(), new FujitsuHeatpumpIR(),
-                            new MitsubishiFDHeatpumpIR(), new MitsubishiFEHeatpumpIR(), new MitsubishiMSYHeatpumpIR(), new MitsubishiFAHeatpumpIR(),
-                            new MitsubishiKJHeatpumpIR(), new MitsubishiHeavyFDTCHeatpumpIR(),
-                            new SamsungAQVHeatpumpIR(), new SamsungFJMHeatpumpIR(),new SharpHeatpumpIR(), new DaikinHeatpumpIR(),
-                            new MitsubishiHeavyZJHeatpumpIR(), new MitsubishiHeavyZMHeatpumpIR(),
-                            new MitsubishiSEZKDXXHeatpumpIR(), new MitsubishiMSCHeatpumpIR(),
-                            new HyundaiHeatpumpIR(), new HisenseHeatpumpIR(),
-                            new GreeGenericHeatpumpIR(), new GreeYANHeatpumpIR(), new GreeYAAHeatpumpIR(),
-                            new FuegoHeatpumpIR(), new ToshibaHeatpumpIR(), new ToshibaDaiseikaiHeatpumpIR(),
-                            new IVTHeatpumpIR(), new HitachiHeatpumpIR(),
-                            new BalluHeatpumpIR(), new AUXHeatpumpIR(),
-                            NULL};
+#include <HeatpumpIRFactory.h>
 
+HeatpumpIR *heatpumpIR;
 IRSenderIRremoteESP8266 *Plugin_088_irSender;
```

Comment:
Always initialize a pointer to `nullptr` as there is no guarantee the compiler will set it to 0/NULL/nullptr.

Reply:
Fixed. heatpumpIR moved into local scope and initialized / set to NULL on delete. Plugin_088_irSender initialized to NULL.